mobile/status.go

```diff
@@ -1046,9 +1045,9 @@ func GetConnectionStringForBeingBootstrapped(configJSON string) string {
 		return makeJSONResponse(fmt.Errorf("no config given, PayloadSourceConfig is expected"))
 	}
 
-	statusBackend.SetLocalPairing(true)
+	statusBackend.LPM().SetPairing(true)
```

Comment:
Now it looks like we save some state in `PairingStateManager` and have the actual state of the pairing Client/Server. And these states are synced only inside these top-level functions in `mobile/status.go`.

I think that now we could create a top-level class `ProcessStateManager` (I'd call it `LocalPairer`, but I don't think there's such a word 😄) and incapsulate this logic in it.

```go
type LocalPairingManager struct {
    backend          *api.GethStatusBackend
    stateManager *pairing.ProcessStateManager
}

func (m *LocalPairingManager) GetConnectionStringForBeingBootstrapped(configJSON) string {
	m.stateManager.SetPairing(true)
	defer func() {
		m.stateManager.SetPairing(false)
	}()
	cs, err := pairing.StartUpReceiverServer(m.backend, configJSON)
	if err != nil {
		return makeJSONResponse(err)
	}
	err = m.backend.Logout()
	if err != nil {
		return makeJSONResponse(err)
	}
	return cs
}
```

We could create `ProcessStateManager` in geth backend, and in `mobile/status.go` we would just call this function. Same for other 3 functions.

Having this done, we would be able to create reliable dependencies between `Server` and `StateManager`. For example, change state with this server timeout callback.
https://github.com/status-im/status-go/blob/ff0628c23b12166e21d95b58d9a7abf696fd902a/server/server.go#L79-L84

_____

This would allow us to:
- Show errors like `This sync code has expired`.
- https://github.com/status-im/status-go/issues/3423

Reply:
> Show errors like `This sync code has expired`

We have countdown screen (displaying to a user when the connection is going to expire) on the client side, if we move that timer to status-go (which is good) then we need to emit signal from status-go updating client how many seconds remain till the expiration, cause otherwise is bad to have 2 counters for the same thing especially that their counters may differ.

Reply:
Yes, I agree and `LocalPairer` sounds like a fine name to me. In fact I was mulling this very idea over as I was implementing this functionality. I can also make an issue for this, as state management of Local pairing and local pairing management should be coupled.

Reply:
> We have countdown screen on the client side

@saledjenic you're talking about sender-side. I was talking about the receiver, if someone scans an expired code, the server would return an "Sync code expired" error. 

Though now I think this might be impossible, because the server shuts down after timeout, so you won't be able to connect to a server with expired connection string.

Reply:
`LocalPairer` issue https://github.com/status-im/status-go/issues/3758

Reply:
`expiresAt` `ConnectParams` string issue https://github.com/status-im/status-go/issues/3759